community/dbms/src/main/java/org/neo4j/commandline/dbms/MemoryRecommendationsCommand.java

```diff
@@ -0,0 +1,265 @@
+/*
+ * Copyright (c) 2002-2018 "Neo Technology,"
+ * Network Engine for Objects in Lund AB [http://neotechnology.com]
+ *
+ * This file is part of Neo4j.
+ *
+ * Neo4j is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.neo4j.commandline.dbms;
+
+import java.util.Locale;
+import java.util.function.ToDoubleFunction;
+
+import org.neo4j.commandline.admin.AdminCommand;
+import org.neo4j.commandline.admin.IncorrectUsage;
+import org.neo4j.commandline.admin.OutsideWorld;
+import org.neo4j.commandline.arguments.Arguments;
+import org.neo4j.commandline.arguments.OptionalNamedArg;
+import org.neo4j.kernel.impl.util.OsBeanUtil;
+
+import static java.lang.String.format;
+import static org.neo4j.configuration.ExternalSettings.initialHeapSize;
+import static org.neo4j.configuration.ExternalSettings.maxHeapSize;
+import static org.neo4j.graphdb.factory.GraphDatabaseSettings.pagecache_memory;
+import static org.neo4j.io.ByteUnit.ONE_GIBI_BYTE;
+import static org.neo4j.io.ByteUnit.ONE_KIBI_BYTE;
+import static org.neo4j.io.ByteUnit.ONE_MEBI_BYTE;
+import static org.neo4j.io.ByteUnit.gibiBytes;
+import static org.neo4j.io.ByteUnit.mebiBytes;
+import static org.neo4j.io.ByteUnit.tebiBytes;
+import static org.neo4j.kernel.configuration.Settings.BYTES;
+import static org.neo4j.kernel.configuration.Settings.buildSetting;
+
+public class MemoryRecommendationsCommand implements AdminCommand
+{
+    // Fields: {System Memory in GiBs; OS memory reserve in GiBs; JVM Heap memory in GiBs}.
```

Comment:
Note that Netty allocates off-heap RAM as well. By default plans to use (`(maxDirectMemory - heap) / 2`); however

- there is no actual upper bound, it will allocate until the malloc fails
- it is lazy, it allocates 16MiB chunks in "arenas:, by default there are `numCpu * 2` arenas, all with one chunk each

TL;DR: Assume Netty allocates 16 * numCPU * 2 MiB at bootstrap, and then unbound additional amounts until we modify the Causal Cluster and Bolt input code to use fixed size I/O buffers.
-